### ISSUE TITLE
learning rate decay_step for D/E and question about D_loss

### ISSUE THREAD

**Issue by doutzi** [Hsintien-Ng/idinvert_pytorch-reproduced#2]
In the original code, the decay_step for the discriminator lr and the encoder lr is set to 30000. Did you set it to 3000 for a reason or is it just a typo?
Another question concerning the D_loss: what is the idea behind _loss_ep_weight * (loss_real * loss_real)_ (line 146), I can't spot this in the original code?

**Comment by Hsintien-Ng** [comment]
Since I found that after 3000 steps the loss did not drop, I change the decay_step to be 3000. It is a super parameter that you can adjust it according to your dataset.

**Comment by Hsintien-Ng** [comment]
loss_ep is the loss that I add to prevent the D from collapse, because D tended to output a large value when training. But now I think it might be because I forgot to add zero_grad(). That's a low-level error. Earlier I claimed that the learning rate needs to be 1e-5, but now it seems that 1e-4 is also possible. Thank you for your review.

**Comment by doutzi** [comment]
Thank you for the clarification.
